### PR TITLE
[master] class cleanup

### PR DESCRIPTION
**Pull request by mitchbregs**
Only partial... need to clean up fetch stats

### CODE REVIEW

File: src/league.py [hunk 1]
```diff
@@ -20,85 +20,94 @@ class League(object):
...
 
-        self.status = resp.status_code
-        request_status(self.status)
+        members = [(x['id'], x['displayName']) for x in data['members']]
+        teams = [(x['id'], x['owners'][0], x['nickname']) for x in data['teams']]
 
-        teams = resp.json()['teams']
+        return {
+            'current_week': data['status']['currentMatchupPeriod'],
+            'nba_day': data['status']['latestScoringPeriod'],
```
Comment (leonkozlowski) [inline src/league.py]: Why remove 'current_week' and 'nba_day' as instance attributes?
having things like `self.current_week` and `self.nba_day` will make dashboard easier in long run- no?

File: src/league.py [hunk 2]
```diff
@@ -20,85 +20,94 @@ class League(object):
...
-        request_status(self.status)
+        # raise error if bad response
+        teams = resp.json()['teams']
 
-        team_data = resp.json()
+        ## fantasy team ids start at 1 not 0
+        my_team = teams[team_id-1]
+
+        team_meta = {
+            'id': team_id,
```
Comment (leonkozlowski) [inline src/league.py]: We needed this metadata!

File: src/league.py [hunk 3]
```diff
@@ -20,85 +20,94 @@ class League(object):
...
-        rosters = resp.json()['teams']
+        ## make more efficient
+        rosters = {}
+        for team in data['teams']:
+            rosters[team['id']] = []
+            for player in team['roster']['entries']:
+                rosters[team['id']].append((
+                    player['playerId'],
+                    player['playerPoolEntry']['player']['fullName']
+                ))
```
Comment (leonkozlowski) [inline src/league.py]: Why make this an array rather than 2 key value pairs?

File: src/league.py (hunk 1 above)
Comment (mitchbregs) [reply src/league.py]: Yea... come to think of it, I agree.. If thats the case, can we add methods that are something like

`def __week___(self)` and in `def init(self): self.week = self.__week__()`.. we'd rather have that hidden...

Totally agree with you thought; my approach was to remove redundancy... it also screws up the mTeams or whatever call parameter.  I'll add that back

File: src/league.py (hunk 3 above)
Comment (mitchbregs) [reply src/league.py]: What do you mean?  The result becomes something like:

```

File: src/league.py (hunk 2 above)
Comment (mitchbregs) [reply src/league.py]: We have it, what do you mean?

File: src/league.py (hunk 3 above)
Comment (leonkozlowski) [reply src/league.py]: Fair enough - I missed the tuple structure for `player_id` and `full_name` was thinking accessing these would get a bit wonky if it was an list of lists.

To be clear its up to preference but accessing (IMO) can be more explicit:
Up to you

```bash
{
    1: [
            {
                'id': 6583, 
                'fullName'': Anthony Davis'
            },
            {
                'id': 6450,
                'fullName': 'Kawhi Leonard'
            }
        ]
}
```

File: src/league.py (hunk 1 above)
Comment (leonkozlowski) [reply src/league.py]: +1 to that comment

Comment (mitchbregs) [reply src/league.py]: Discussed IRL, going to import meta at initialization

File: src/league.py [hunk 4]
```diff
@@ -113,13 +116,14 @@ def _fetch_rosters(self, team_id=None):
         for team in data['teams']:
             rosters[team['id']] = []
             for player in team['roster']['entries']:
-                rosters[team['id']].append((
-                    player['playerId'],
-                    player['playerPoolEntry']['player']['fullName']
-                ))
+                rosters[team['id']].append({
```
Comment (leonkozlowski) [inline src/league.py]: Nice

**Pull request merged**